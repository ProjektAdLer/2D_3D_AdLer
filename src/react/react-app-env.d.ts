/// <reference types="react-scripts" />

// Images
declare module '*.jpg';
declare module '*.png';
declare module '*.env';

// 3D types
declare module '*.glb';
declare module '*.stl';

declare module '*.jpeg';
declare module '*.gif';
declare module '*.mp4';
