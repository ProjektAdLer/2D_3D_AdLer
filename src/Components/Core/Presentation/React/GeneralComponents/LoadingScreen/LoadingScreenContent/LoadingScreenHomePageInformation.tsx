import { AdLerUIComponent } from "src/Components/Core/Types/ReactTypes";
import adlerHomepage from "../../../../../../../Assets/misc/AdLerHomepage.jpg";
import tailwindMerge from "src/Components/Core/Presentation/Utils/TailwindMerge";
import StyledButton from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";
import { useTranslation } from "react-i18next";

export default function LoadingScreenHomePageInformation({
  className,
}: Readonly<AdLerUIComponent<{}>>) {
  const { t: translate } = useTranslation("loadingScreen");

  return (
    <div className={tailwindMerge(className, "grid grid-cols-10")}>
      <h1 className="col-span-6 col-start-3 p-2 font-bold text-center portrait:pb-6 portrait:col-span-10 portrait:col-start-1 text-adlerdarkblue pd-4">
        {translate("websiteInfoText")}
      </h1>
      <StyledButton
        shape="freeFloatCenterNoPadding"
        containerClassName="col-span-4 col-start-4 p-2 portrait:col-span-8 portrait:col-start-2"
        feedback="nothing"
      >
        <div className="bg-slate-400">
          <a
            href="https://projekt-adler.eu"
            target="_blank"
            rel="noreferrer"
            className="opacity-75 hover:opacity-100"
          >
            <img src={adlerHomepage} alt="adlerHomepage" />
          </a>
        </div>
      </StyledButton>
    </div>
  );
}
