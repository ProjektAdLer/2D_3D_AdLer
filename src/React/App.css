body {
  margin: 0;
  height: 100vh;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

div{
  width: 100%;
  height: 100%;
}

section{
  display: inline;
  position: fixed;
  top: 0;
  left: 0;
  width: 120px;
  height: 100vh;
  background-color: #fff;
}

button{
  width: 100px;
  margin: 10px;
  z-index: 5;
  position: fixed;
  left: 0;
  top: 0;
}

canvas {
  width: calc(100% - 100px);
  height: 100%;
  margin-left: 100px;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.modal {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
}

.modal-content {
  width: 1000px;
  background-color: #fff;
}

.modal-header,
.modal-footer {
  padding: 10px;
  height: 60px;
}

.modal-title {
  margin: 0;
}

.modal-body {
  padding: 10px;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
}
