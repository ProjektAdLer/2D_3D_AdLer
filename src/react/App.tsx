import React from 'react';
import './App.css';
import BabylonCanvas from './components/BabylonCanvas';

function App() {
	return (
		<React.Fragment>
			<BabylonCanvas style={{ width: '100%', position: 'absolute' }} />
		</React.Fragment>
	);
}

export default App;
