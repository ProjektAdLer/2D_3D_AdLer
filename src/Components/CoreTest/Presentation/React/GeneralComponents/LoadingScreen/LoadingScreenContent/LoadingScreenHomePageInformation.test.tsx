import React from "react";
import { render } from "@testing-library/react";
import LoadingScreenHomePageInformation from "../../../../../../Core/Presentation/React/GeneralComponents/LoadingScreen/LoadingScreenContent/LoadingScreenHomePageInformation";

describe("LoadingScreenHomePageInformation", () => {
  test("should render", () => {
    const { container } = render(<LoadingScreenHomePageInformation />);
    expect(container).toMatchSnapshot();
  });
});
