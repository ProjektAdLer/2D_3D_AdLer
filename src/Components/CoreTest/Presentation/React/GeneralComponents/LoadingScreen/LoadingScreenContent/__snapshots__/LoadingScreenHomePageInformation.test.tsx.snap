// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreenHomePageInformation should render 1`] = `
<div>
  <div
    class="grid grid-cols-10"
  >
    <h1
      class="col-span-6 col-start-3 p-2 font-bold text-center portrait:pb-6 portrait:col-span-10 portrait:col-start-1 text-adlerdarkblue pd-4"
    >
      websiteInfoText
    </h1>
    <div
      class="col-span-4 col-start-4 p-2 portrait:col-span-8 portrait:col-start-2"
    >
      <button
        class="justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
      >
        <div
          class="bg-slate-400"
        >
          <a
            class="opacity-75 hover:opacity-100"
            href="https://projekt-adler.eu"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="adlerHomepage"
              src="AdLerHomepage.png"
            />
          </a>
        </div>
      </button>
    </div>
  </div>
</div>
`;
